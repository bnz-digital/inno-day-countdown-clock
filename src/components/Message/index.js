import React from 'react'
import { not } from 'ramda'
import { object, oneOfType, string } from 'prop-types'
import { isNilOrEmpty } from 'ramda-adjunct'

import StyledMessage from './styled'
import StyledMain from '../styled/Main'

export default function Message ({ message }) {
  return (
    not(isNilOrEmpty(message)) &&
    <StyledMain>
      <StyledMessage>{message}</StyledMessage>
    </StyledMain>
  )
}

Message.propTypes = {
  message: oneOfType([object, string])
}
