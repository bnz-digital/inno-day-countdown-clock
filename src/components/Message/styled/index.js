import styled from 'styled-components'

const StyledMessage = styled.div`
  font-size: 4vw;
  grid-area: message;
  place-self: center;
  padding-top: 10vh;
`
StyledMessage.displayName = 'StyledMessage'

export default StyledMessage
