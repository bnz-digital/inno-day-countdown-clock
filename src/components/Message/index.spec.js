import React from 'react'
import { shallow } from 'enzyme'

import Message from '.'

describe('components:Message', () => {
  it('renders <Message/> correctly with component message', () => {
    expect(
      toJson(shallow(<Message message={<p>hello</p>} />))
    ).toMatchSnapshot()
  })

  it('renders <Message/> correctly with string message', () => {
    expect(toJson(shallow(<Message message='howdy' />))).toMatchSnapshot()
  })

  it('renders <Message/> correctly without message', () => {
    expect(toJson(shallow(<Message />))).toMatchSnapshot()
  })
})
