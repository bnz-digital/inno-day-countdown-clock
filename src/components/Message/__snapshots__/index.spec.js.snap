// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Message renders <Message/> correctly with component message 1`] = `
<StyledMain>
  <StyledMessage>
    <p>
      hello
    </p>
  </StyledMessage>
</StyledMain>
`;

exports[`components:Message renders <Message/> correctly with string message 1`] = `
<StyledMain>
  <StyledMessage>
    howdy
  </StyledMessage>
</StyledMain>
`;

exports[`components:Message renders <Message/> correctly without message 1`] = `""`;
